### PR TITLE
[Dev] replace changelog script in dev with custom approach

### PR DESCRIPTION
**Pull request by merobi-hub**
### Problem

The existing script in dev for partially automating changelog updates relies on a fork of an external tool and does not format additions correctly or supply all the details needed.

### Solution

This replaces the script with a new Python class for getting all commits and pulls since a specified release, except for bot version updates, and adding change descriptions to the changelog. It also supplies a release title along with all URLs and one-line descriptions from PRs (if provided by authors) in the format we have been using. Finally,  `check_new_contributor` and `print_new_contributors` methods identify new contributors based on welcome comments from the boring-cyborg bot and print them to the terminal. The readme and requirements.txt have been updated, as well.

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [x] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

File: dev/get_changes.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+#
+# Copyright 2018-2023 contributors to the OpenLineage project
+# SPDX-License-Identifier: Apache-2.0
+
+from github import Github
+import rich_click as click
```
Comment (mobuchowski) [inline dev/get_changes.py:7]: Please add requirements.txt if you're importing external libraries

Comment (merobi-hub) [reply dev/get_changes.py:7]: Thanks @mobuchowski . The requirements.txt in dev has been updated.

File: dev/README.mdx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+### Using `get_changes`
```
Comment (mobuchowski) [inline dev/README.mdx:1]: why `README.mdx`? Is that a typo or should it be like that?

File: dev/get_changes.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+@click.option(
+    "--previous", type=str, default=''
+)
+@click.option(
+    "--current", type=str, default=''
+)
+@click.option(
+    "--path", 
+    type=str, 
+    default='/Users/michael/Documents/astronomer_projects/openlineage_upstream/OpenLineage/CHANGELOG.md',
```
Comment (mobuchowski) [inline dev/get_changes.py:110]: 🙂

Comment (mobuchowski) [reply dev/get_changes.py:110]: Can we make this relative path?

Comment (merobi-hub) [reply dev/get_changes.py:110]: Unfortunately no -- relative paths aren't supported by the library. But I can remove the default.

File: dev/README.mdx (hunk 2 above)
Comment (merobi-hub) [reply dev/README.mdx:1]: Not a typo, but MDX is probably overkill. MD will work, and I will change it back.

**Review by julienledem — APPROVED** [review]
That makes sense to me. @mobuchowski is this looking good to you as well?
We'd need this merged for the next release.

File: dev/get_changes.py (hunk 3 above)
Comment (merobi-hub) [reply dev/get_changes.py]: Despite the docs saying an absolute path is required, a relative path works. I tested it and made it the default.

**Comment by mobuchowski** [comment]
@merobi-hub want to merge this?

**Pull request merged**